From f406ddef87d1f090e05659e91add7d0df6ea0c2e Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Thu, 3 Dec 2015 18:07:13 +0800
Subject: [PATCH 0758/5242] MLK-11957 input: misc: fxls8471: remove
 unnecessary .kfree()

commit  21e40304c66f5c3fb6c172ba7fd2f3b674300887 from
https://source.codeaurora.org/external/imx/linux-imx.git

The driver kfree the global memory that is not correct. The patch
remove them.

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/input/misc/fxls8471.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/input/misc/fxls8471.c b/drivers/input/misc/fxls8471.c
index 0b4b005..e77ef1b 100644
--- a/drivers/input/misc/fxls8471.c
+++ b/drivers/input/misc/fxls8471.c
@@ -496,7 +496,7 @@ int fxls8471_driver_init(struct fxls8471_data *pdata)
 	result = misc_register(&fxls8471_device);
 	if (result != 0) {
 		printk(KERN_ERR "register acc miscdevice error");
-		goto err_regsiter_misc;
+		goto err_out;
 	}
 
 	result =
@@ -548,8 +548,6 @@ int fxls8471_driver_init(struct fxls8471_data *pdata)
 			   &fxls8471_attr_group);
 err_create_sysfs:
 	misc_deregister(&fxls8471_device);
-err_regsiter_misc:
-	kfree(pdata);
 err_out:
 	return result;
 }
@@ -559,8 +557,7 @@ int fxls8471_driver_remove(struct fxls8471_data *pdata)
 {
 	fxls8471_change_mode(pdata, STANDBY);
 	misc_deregister(&fxls8471_device);
-	if (pdata != NULL)
-		kfree(pdata);
+
 	return 0;
 }
 EXPORT_SYMBOL_GPL(fxls8471_driver_remove);
-- 
1.7.9.5

